code/modules/mob/living/carbon/xenomorph/Abilities.dm

```diff
@@ -1094,10 +1083,8 @@ datum/action/xeno_action/activable/salvage_plasma/improved
 		new_xeno.queen_chosen_lead = TRUE
 		new_xeno.hud_set_queen_overwatch()
 
-	var/datum/hive_status/hive = hive_datum[X.hivenumber]
-
-	if(hive.living_xeno_queen && hive.living_xeno_queen.observed_xeno == T)
-		hive.living_xeno_queen.set_queen_overwatch(new_xeno)
+	if(X.hive.living_xeno_queen && X.hive.living_xeno_queen.observed_xeno == T)
```

Comment:
> ?
```suggestion
	if(X.hive.living_xeno_queen?.observed_xeno == T)
```

Reply:
While this will work it's best to avoid combining checking if something exists to comparison. As you may get funky values with things like null you didn't expect.

Reply:
You mean when `T` is `null` ? Indeed, good point.